modules/ting/src/Search/TingSearchFacet.php

```diff
@@ -0,0 +1,102 @@
+<?php
+/**
+ * @file
+ * The TingSearchFacet class.
+ */
+
+namespace Ting\Search;
+
+/**
+ * Represents a Search facet and it's terms.
+ *
+ * @package Ting\Search
+ */
+class TingSearchFacet {
+
+  /**
+   * @var string
+   */
+  protected $name;
+
+  /**
+   * Terms in the facet.
+   *
+   * Used when the facet is a part of a search result.
+   *
+   * The list is keyed by term name.
+   *
+   * @var \Ting\Search\TingSearchFacetTerm[]
+   */
+  protected $terms = [];
+
+  /**
+   * TingSearchFacet constructor.
+   *
+   * @param string $name
+   *   Name of the facet.
+   *
+   * @param TingSearchFacetTerm[] $terms
+   *   Optional list of terms.
+   */
+  public function __construct($name, $terms = []) {
+    $this->name = $name;
+    // Set via setter to store the terms by their names.
+    $this->setTerms($terms);
+  }
+
+  /**
+   * Gets the name of the facet.
+   *
+   * @return string
+   *   The name.
+   */
+  public function getName() {
+    return $this->name;
+  }
+
+  /**
+   * Sets the name of the facet.
+   *
+   * @param string $name
+   *   The name.
+   */
+  public function setName($name) {
```

Comment:
I know this is a topic which has not been discussed in the other PRs but I wondered: Does it make sense to have setters for these properties? I mean: Should we allow changing these once they have been defined when constructing the object?

Reply:
And it is a good thing to keep in mind in general. I agree, I don't see the usecase for changing the name of the facet.

Reply:
👍🏾